pkg/agent/flowexporter/connections/conntrack_linux.go

```diff
@@ -0,0 +1,192 @@
+		// Only get Pod-to-Pod flows. Pod-to-ExternalService flows are ignored for now.
+		if srcIP.Equal(ctdump.nodeConfig.GatewayConfig.IP) || dstIP.Equal(ctdump.nodeConfig.GatewayConfig.IP) {
+			continue
+		}
+
+		// Pod-to-Service flows w/ kube-proxy: There are two conntrack flows for every Pod-to-Service flow.
+		// One is with ClusterIP as source or destination, where other IP is podIP. Second conntrack flow is
+		// with resolved Endpoint Pod IP corresponding to ClusterIP. Both conntrack flows have same stats, which makes them duplicate.
+		// Ideally, we have to correlate these two connections and maintain one connection with both Endpoint Pod IP and ClusterIP.
+		// To do the correlation, we need ClusterIP-to-EndpointIP mapping info, which is not available at Agent.
+		// Therefore, we ignore the connection with ClusterIP and keep the connection with Endpoint Pod IP.
+		// Conntrack flows will be different for Pod-to-Service flows w/ Antrea-proxy. This implementation will be simpler, when the
+		// Antrea proxy is supported.
+		if ctdump.serviceCIDR.Contains(srcIP) || ctdump.serviceCIDR.Contains(dstIP) {
+			continue
+		}
+		filteredConns = append(filteredConns, createAntreaConn(&conn))
+	}
+
+	klog.V(2).Infof("Finished poll cycle -- total flows: %d flows in Antrea zoneID: %d", len(conns), len(filteredConns))
+
+	return filteredConns, nil
+}
+
+// connTrackSystem implements ConnTrackInterfacer
+var _ ConnTrackInterfacer = new(connTrackSystem)
+
+// ConnTrackInterfacer is an interface created to consume the required functions from the third party
+// conntrack library. This is helpful in writing unit tests.
+type ConnTrackInterfacer interface {
+	Dial() error
+	DumpFilter(filter conntrack.Filter) ([]conntrack.Flow, error)
+}
+
+type connTrackSystem struct {
+	netlinkConn *conntrack.Conn
+}
+
+func NewConnTrackInterfacer() *connTrackSystem {
+	// Check value of setting net.netfilter.nf_conntrack_acct. Set to value 1, if it is not set.
+	connTrackAcct, err := sysctl.GetSysctlNet("netfilter/nf_conntrack_acct")
+	if err != nil {
+		// Continue with creation of interfacer object as we can dump flows with no stats and that information can still be useful.
+		// If permission error, please provide access to net.netfilter.nf_conntrack_acct. This will enable flow exporter to export stats and timestamps of connections.
+		klog.Errorf("Error when getting net.netfilter.nf_conntrack_acct: %v", err)
+	} else {
+		if connTrackAcct == 0 {
+			err = sysctl.SetSysctlNet("netfilter/nf_conntrack_acct", 1)
+			if err != nil {
+				// If permission error, please provide access to net.netfilter.nf_conntrack_acct.
+				klog.Errorf("Error when setting net.netfilter.nf_conntrack_acct: %v", err)
+			}
+			// Set the conntrack timestamp setting to get timestamps of connections
+			err = sysctl.SetSysctlNet("netfilter/nf_conntrack_timestamp", 1)
+			if err != nil {
+				klog.Errorf("Error when setting net.netfilter.nf_conntrack_timestamp: %v", err)
+			}
+		}
+	}
```

Comment:
This new code is organized a bit strangely I think.
Could we define an `ensureNfConntrackFlag(flagName string, value int)` function, and then call it like this here:
```
ensureNfConntrackFlag("nf_conntrack_acct", 1)
ensureNfConntrackFlag("nf_conntrack_timestamp", 1)
```

You used to have some special handling for permission errors but now you just have these comments: "If permission error, please provide access to net.netfilter.nf_conntrack_acct". I find it a bit confusing. Shouldn't we have this in a log message so the user can see it?

Reply:
Yes, will change the organization accordingly. 
@permisssion error: I realized that error already says permission access issue, so I removed from log. In addition, for other errors, I felt we do not need to return error as the rest of the flow data other than stats and timestamp is still useful.

Reply:
I considered both conntrack settings to be independent in latest commit.

Reply:
The code looks better, but I stand by my earlier comment: since you do the same thing for both settings, why not define a helper function (e.g. `ensureNfConntrackFlag`) which takes the settings name as a parameter?

Reply:
No strong opinion here. Added helper function. 
In prior way comments in the code for each setting are right before the error logs; felt that would be easier to read rather than having them in the callee of helper func.